packages/wonder-blocks-icon-button/src/components/activity-icon-button.tsx

```diff
@@ -0,0 +1,261 @@
+ *
+ * <ActivityIconButton
+ *     icon={magnifyingGlassIcon}
+ *     aria-label="An Icon"
+ *     onClick={(e) => console.log("Hello, world!")}
+ * />
+ * ```
+ */
+export const ActivityIconButton: React.ForwardRefExoticComponent<
+    Props &
+        React.RefAttributes<typeof Link | HTMLButtonElement | HTMLAnchorElement>
+> = React.forwardRef<
+    typeof Link | HTMLButtonElement | HTMLAnchorElement,
+    Props
+>(function ActivityIconButton(props: Props, ref) {
+    const {
+        actionType = "progressive",
+        disabled = false,
+        icon,
+        kind = "primary",
+        style,
+        type = "button",
+        ...restProps
+    } = props;
+
+    const [pressed, setPressed] = React.useState(false);
+
+    const buttonStyles = _generateStyles(actionType, !!disabled, kind);
+
+    const styles = [
+        buttonStyles.button,
+        disabled && buttonStyles.disabled,
+        !disabled && pressed && buttonStyles.pressed,
+        style,
+    ];
+
+    const chonkyStyles = [
+        buttonStyles.chonky,
+        disabled && buttonStyles.chonkyDisabled,
+        !disabled && pressed && buttonStyles.chonkyPressed,
+    ];
+
+    const handlePress = React.useCallback((isPressing: boolean) => {
+        setPressed(isPressing);
+    }, []);
+
+    return (
+        <IconButtonUnstyled
+            {...restProps}
+            disabled={disabled}
+            onPress={handlePress}
+            ref={ref}
+            style={styles}
+            type={type}
+        >
+            {/* NOTE: Using a regular className to be able to use descendant selectors to account for the hover and press states */}
+            <View style={chonkyStyles} className="chonky">
+                <PhosphorIcon size="medium" color="currentColor" icon={icon} />
+            </View>
+        </IconButtonUnstyled>
+    );
+});
+
+// NOTE: Theme is defined in the file directly to avoid generating CSS variables
+// as we are reusing the following tokens in all themes.
+const theme = {
+    root: {
+        border: {
+            width: {
+                primary: {
+                    rest: border.width.thin,
+                    hover: border.width.thin,
+                    press: border.width.thin,
+                },
+                secondary: {
+                    rest: border.width.thin,
+                    hover: border.width.thin,
+                    press: border.width.thin,
+                },
+                tertiary: {
+                    rest: border.width.thin,
+                    hover: border.width.thin,
+                    press: border.width.thin,
+                },
+            },
+            radius: border.radius.radius_120,
+        },
+        layout: {
+            padding: {
+                block: sizing.size_140,
+                inline: sizing.size_200,
+            },
+        },
+        shadow: {
+            y: {
+                rest: "6px",
+                hover: "8px",
+                press: sizing.size_0,
```

Comment:
Would it make sense to have the shadow `y` values as tokens as well? (for later!)

And is there a reason why we use `px` instead of `rem`? It would be helpful to include a comment here so we remember that it is intentional!